src/main/java/nextstep/path/service/PathFinder.java

```diff
@@ -15,24 +20,36 @@ public class PathFinder {
     private StationService stationService;
     private LineService lineService;
     private PathService pathService;
+    private MemberService memberService;
+    private CalculateFareService calculateFareService;
 
-    public PathFinder(StationService stationService, LineService lineService, PathService pathService) {
+    public PathFinder(StationService stationService, LineService lineService, PathService pathService, MemberService memberService, CalculateFareService calculateFareService) {
         this.stationService = stationService;
         this.lineService = lineService;
         this.pathService = pathService;
+        this.memberService = memberService;
+        this.calculateFareService = calculateFareService;
     }
 
     @Transactional(readOnly = true)
-    public PathResponse retrieveStationPath(final String type, final Long source, final Long target) {
+    public PathResponse retrieveStationPath(final LoginMember loginMember, final String type, final Long source, final Long target) {
         validateStationExist(source, target);
         List<Line> lineList = lineService.getAllLines();
-        return pathService.findPath(type, source, target, lineList);
+        Member member = findMemberByOptionalLoginMember(loginMember);
+        Path path = pathService.findPath(member, type, source, target, lineList);
+        calculateFareService.calculateFare(path);
```

Comment:
```suggestion
        Path path = pathService.findPath(member, type, source, target, lineList);
        Long fare = calculateFareService.calculateFare(member, price, someArgument...);
        return PathResponse.of(path.xxx, fare.xxx);
```
Path가 반드시 요금 정보를 가지고 있지 않아도 된다고 생각해요. 그래서 위와 같이 path, fare를 따로 구한 뒤 응답 객체에 담아주는 방식도 고민해보셨으면 해요.

Reply:
아 그런 방법이 더 적절하겠네요! 고민했는데 감사합니다!!